service/history/api/create_workflow_util.go

```diff
@@ -86,6 +91,23 @@ func NewWorkflowWithSignal(
 		return nil, err
 	}
 
+	if startRequest.StartRequest.WorkflowType.Name == scheduler.WorkflowType && shard.GetConfig().UseExperimentalHsmScheduler(startRequest.NamespaceId) {
```

Comment:
Just a note for other reviewers, we're aware this is a hack and we'll clean it up after we finish this PoC.